RxSwift/Observables/TakeUntil.swift

```diff
@@ -129,3 +156,72 @@ final fileprivate class TakeUntil<Element, Other>: Producer<Element> {
         return (sink: sink, subscription: subscription)
     }
 }
+
+// MARK: - TakeUntil Predicate
+final fileprivate class TakeUntilPredicateSink<O: ObserverType>
+    : Sink<O>, ObserverType {
+    typealias Element = O.E
+    typealias Parent = TakeUntilPredicate<Element>
+
+    fileprivate let _parent: Parent
+    fileprivate var _running = true
+
+    init(parent: Parent, observer: O, cancel: Cancelable) {
+        _parent = parent
+        super.init(observer: observer, cancel: cancel)
+    }
+
+    func on(_ event: Event<Element>) {
+        switch event {
+        case .next(let value):
+            if !_running {
```

Comment:
I would probably put `self` everywhere. I know that we weren't consistent in the past, but looks good enough to pull in.

Reply:
I’ll  take care of this in a separate PR :) thanks !